adapters/dmx/dmx.go

```diff
@@ -0,0 +1,245 @@
+package dmx
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbs"
+	"net/http"
+	"sort"
+)
+
+type DmxAdapter struct {
+	endpoint    string
+	publisherId string
+}
+
+func New(text string) error {
+	return &errorString{text}
+}
+
+// errorString is a trivial implementation of error.
+type errorString struct {
+	s string
+}
+
+func (e *errorString) Error() string {
+	return e.s
+}
+
+func NewDmxBidder(endpoint string, publisher_id string) *DmxAdapter {
+	return &DmxAdapter{endpoint: endpoint, publisherId: publisher_id}
+}
+
+type dmxBidder struct {
+	Bidder dmxExt `json:"bidder"`
+}
+
+type dmxUser struct {
+	User *openrtb.User `json:"user"`
+}
+
+type dmxExt struct {
+	PublisherId string `json:"publisher_id,omitempty"`
+}
+
+type dmxBanner struct {
+	Banner *openrtb.Banner `json:"banner"`
+}
+
+type dmxSize struct {
+	W uint64
+	H uint64
+	S uint64
+}
+
+type DmxSize []dmxSize
+
+func (a DmxSize) Len() int {
+	return len(a)
+}
+
+func (a DmxSize) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+
+func (a DmxSize) Less(i, j int) bool {
+	return a[i].S > a[j].S
+}
+
+func Remove(toBeRemove []openrtb.Format, a DmxSize) (dmx DmxSize) {
+	for _, value := range toBeRemove {
+		for _, dmxValue := range a {
+			if dmxValue.H == value.H && dmxValue.W == value.W {
+				//fmt.Println("true")
+				dmx = append(dmx, dmxValue)
+			}
+		}
+	}
+	return
+}
+
+var CheckTopSizes = []dmxSize{
+	{300, 250, 100},
+	{728, 90, 95},
+	{320, 50, 90},
+	{160, 600, 88},
+	{300, 600, 85},
+	{300, 50, 80},
+	{970, 250, 75},
+	{970, 90, 70},
+}
+
+func (adapter *DmxAdapter) MakeRequests(request *openrtb.BidRequest, req *adapters.ExtraRequestInfo) (reqsBidder []*adapters.RequestData, errs []error) {
+	var dmxImp dmxBidder
+	var imps []openrtb.Imp
+	//var userParams *dmxUser
+	if err := json.Unmarshal(request.Imp[0].Ext, &dmxImp); err != nil {
```

Comment:
Please verify there is at least 1 Imp in the collection before indexing it via `request.Imp[0].Ext`.